### ISSUE TITLE
Build only those images consumed by CCI

### ISSUE THREAD

**Issue by uilianries** [conan-io/conan-docker-tools#563]
### Description of Problem, Request, or Question

Building Docker images is consuming several ours in the CI, but the mainly cost is producing images that are not consumed by ConanCenterIndex.

It would be faster if we could build only what is listed in CCI:

- https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml
- https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v2.yml

In summary:

Legacy images: GCC5, GCC7 and GCC9
Modern images: GCC11 and Clang13

**Comment by Croydon** [comment]
@uilianries The docs are outdated then? https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#linux

**Comment by uilianries** [comment]
@Croydon Good catch! Let me update it, I know it's updated by a CI job. Probably failed for some reason. Thank you for pinging!

**Comment by uilianries** [comment]
Done, new hotfix available here https://github.com/conan-io/conan-center-index/pull/22870
